src/core/actions/cityactions/Spawn.java

```diff
@@ -23,19 +28,48 @@ public Spawn(City c)
 
     @Override
     public LinkedList<Action> computeActionVariants(GameState gs) {
-        //TODO: compute spawn actions
-        return null;
+        LinkedList<Action> actions = new LinkedList<>();
+
+        Tribe tribe = gs.getTribe(city.getTribeId());
+        TechnologyTree t = tribe.getTechTree();
+        int star = tribe.getStars();
+        if (gs.getBoard().getUnitAt(city.getX(), city.getY()) == null){
+            for(Types.UNIT unit: Types.UNIT.values()){
+                // Only can create land units
+                if (unit.getCost() <= star && unit.getKey() <= 7){
```

Comment:
Line 39: I think we may have to change that '7' for something different, it's not very flexible @rovenr12